classes/core/PKPPageRouter.php

```diff
@@ -560,11 +560,13 @@ private function _setLocale(PKPRequest $request, ?string $setLocale): void
      */
     private function isAllowedHost(string $protocol, string $redirectUrl): bool
     {
+        $allowedHosts = Config::getVar('general', 'allowed_hosts');
+        if ($allowedHosts == '') { // disabled (null) or empty string ''
+            return true;
+        }
         if (str_starts_with($redirectUrl, $protocol)) {
             $redirectUrl = preg_replace('/^' . preg_quote($protocol . '://', '/') . '/', '', $redirectUrl);
-            ;
         }
-        $allowedHosts = Config::getVar('general', 'allowed_hosts');
         $allowedHosts = array_map(strtolower(...), json_decode($allowedHosts));
```

Comment:
This will still cause issue if someone set the `allowed_hosts` to some string or invalid data as `json_decode` will return `null` . If I understand correctly , `allowed_hosts` should be valid structure which is can be json decoded . Wondering rather than allowing any kind of format which will cause issue, should we force check above if `allowed_hosts` is and also properly json decodable, only then run the array mapping ?  and in that case , probably do error log or perhaps exception(if we want to be very strict) ?

Reply:
I think the AllowedHostsPolicy (https://github.com/pkp/pkp-lib/blob/stable-3_5_0/classes/security/authorization/AllowedHostsPolicy.php) will decline the access to the web site even earlier if the allowed_hosts is some invalid structure that cannot be json decoded. 
However, maybe I can add the check here and log an error if it cannot be json decoded, as you suggested...

Reply:
Hi @touhidurabir, I have just tested and thought about this once again: I think there is no much sense in handling it here -- if there is something wrong in the allowed_hosts, that policy will handle it i.e. the error will pop up earlier. And it is probably not going to happen, that someone just changes this setting manually before changing the UI language.
What do you think?

Reply:
@bozana you are right , we can leave it to that policy .